plugins/catalog/src/plugin.ts

```diff
@@ -30,6 +30,8 @@ import {
   discoveryApiRef,
   fetchApiRef,
   storageApiRef,
+  PluginOptions,
+  PluginInputOptions,
```

Comment:
Realizing that these should be defined in the catalog plugin and be the exact values that the catalog plugin supports. Really important to have TypeScript help out here a show what is available for configuration.

Reply:
Ah hold up, the contents of `src` in `catalog-customized` would pretty much just be the call to `reconfigure`. Imagine it lives in an internal backstage project that want so do some customization of the catalog and keep that separated from the rest of the app code.

Reply:
@Rugvip 
Please have a look at the example created.

I'd prefer to not include Utility API in this PR, but split it:
* it would make sense to start small, then to add full-fledged features
* it's already quite a while in progress
* I'm not sure how much time I'll get in upcoming weeks.

Reply:
Yep fine to skip support for extra things as long as we can ship it all as experimental APIs. Tbh I think that's probably the one thing we need to fix to make this mergeable.

Support for Utility APIs is most likely a requirement for making this stable API though, but we can leave that for the future. Same thing with the interaction that his has with #10256, as there's some significant overlap.